kingpin/actors/aws/cloudformation.py

```diff
@@ -1080,6 +1114,29 @@ def _diff_params_safely(self, remote, local):
 
         return False
 
+    def _template_body_with_hash(self) -> str:
+        """Add a hash to the template to force a change in the stack."""
+
+        # Bail if the user has disabled this feature.
+        if len(KINGPIN_CFN_HASH_OUTPUT_KEY) == 0:
```

Comment:
I will test how python handles env vars and the difference between passing in an empty string vs a "null" (in bash). I thought it would be easier to just keep it as a string and do a length comparison. That way a user could disable the hash-management functionality by passing an empty string where as the default case would be that we do use the hash functionality.

Reply:
My intent is the following:

If env var is:
- a string of length > 0, use it as the output key
- an empty string, disable functionality
- not set or otherwise null, use the default output key

Reply:
Using this simple script here are the results of what I found...

```python
import os

ADK_CONST = os.getenv("ADK_CONST", "adkdefault")

if __name__ == "__main__":
    print(f"ADK_CONST: \"{ADK_CONST}\"")
    print(f"ADK_CONST type: \"{type(ADK_CONST)}\"")
```

```
.venv) akennedy@ndm2a kingpin % python adk.py              
ADK_CONST: "adkdefault"
ADK_CONST type: "<class 'str'>"
(.venv) akennedy@ndm2a kingpin % env ADK_CONST= python adk.py
ADK_CONST: ""
ADK_CONST type: "<class 'str'>"
(.venv) akennedy@ndm2a kingpin % ADK_CONST= python adk.py 
ADK_CONST: ""
ADK_CONST type: "<class 'str'>"
(.venv) akennedy@ndm2a kingpin % env ADK_CONST="" python adk.py
ADK_CONST: ""
ADK_CONST type: "<class 'str'>"
(.venv) akennedy@ndm2a kingpin % ADK_CONST="" python adk.py    
ADK_CONST: ""
ADK_CONST type: "<class 'str'>"
(.venv) akennedy@ndm2a kingpin % env ADK_CONST="adkoverride" python adk.py
ADK_CONST: "adkoverride"
ADK_CONST type: "<class 'str'>"
(.venv) akennedy@ndm2a kingpin % ADK_CONST="adkoverride" python adk.py 
ADK_CONST: "adkoverride"
ADK_CONST type: "<class 'str'>"
```

Reply:
But ... you don't need to do that?
```python
>>> import os
>>> def isTrue(str: str):
...     if (str):
...         print("true")
...     else:
...         print("false")
...         
>>> isTrue(os.getenv("MISSING"))
false
>>> isTrue(os.getenv("USER"))
true
>>> 
```

Reply:
so simply writing: `if KINGPIN_CFN_HASH_OUTPUT_KEY: ...` should work.